Convert from v0 meta.yaml to v1 recipe.yaml syntax

weiji14 opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

Bringing in changes from my PR at https://github.com/conda-forge/staged-recipes/pull/29888

weiji14 committed:
Convert from v0 meta.yaml to v1 recipe.yaml syntax

weiji14 committed:
MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.10.05.34.47

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

weiji14 commented on recipe/meta.yaml:
```diff
@@ -1,64 +0,0 @@
-    - transformers
-    - packaging
-    - setuptools >=61.0.0
-    - causal-conv1d >=1.5.2
```
This dependency is marked as optional in https://github.com/state-spaces/mamba/blob/e0761ece1db07e0949dd88b4f4cd440420a19fd9/pyproject.toml#L30-L33, so I've removed it from the runtime dependency list.

isuruf replied:
Usually conda packages come with batteries included so that users get the best performance.

weiji14 replied:
Mm, I guess it makes sense here. Added it back in commit 50263d6ca9b70b675e6af277f5a4b4789d339dcd

weiji14 replied:
Ah, there's a conflict:

<details>

```
Error:   × Test failed: failed to setup test environment: Cannot solve the request
  │ because of: The following packages are incompatible
  │ └─ mamba-ssm ==2.2.5 py310he9a7d99_1 cannot be installed because there are
  │ no viable options:
  │    ├─ mamba-ssm 2.2.5 would require
  │    │  ├─ pytorch >=2.5.1,<2.6.0a0, which can be installed with any of the
  │ following options:
  │    │  │  ├─ pytorch 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1
  │ | 2.5.1 | 2.5.1 would require
  │    │  │  │  └─ cudatoolkit >=11.8,<12, which can be installed with any of
  │ the following options:
  │    │  │  │     └─ cudatoolkit 11.8.0 | 11.8.0 | 11.8.0 | 11.8.0
  │    │  │  └─ pytorch 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1
  │ | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 |
  │ 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1 | 2.5.1
  │    │  ├─ causal-conv1d >=1.5.2, which cannot be installed because there
  │ are no viable options:
  │    │  │  ├─ causal-conv1d 1.5.2 | 1.5.2 would require
  │    │  │  │  └─ libtorch >=2.7.1,<2.8.0a0, which cannot be installed
  │ because there are no viable options:
  │    │  │  │     └─ libtorch 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 |
  │ 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 | 2.7.1 |
  │ 2.7.1 | 2.7.1 would constrain
  │    │  │  │        └─ pytorch ==2.7.1 cpu_generic_*_0, which conflicts with
  │ any installable versions previously reported
  │    │  │  ├─ causal-conv1d 1.5.2 | 1.5.2 | 1.5.2 | 1.5.2 | 1.5.2 would
  │ require
  │    │  │  │  ├─ python >=3.13,<3.14.0a0, which can be installed with any of
  │ the following options:
  │    │  │  │  │  └─ python 3.13.0 | 3.13.0 | 3.13.1 | 3.13.1 | 3.13.1 |
  │ 3.13.1 | 3.13.1 | 3.13.2 | 3.13.2 | 3.13.3 | 3.13.3 | 3.13.5 | 3.13.5 |
  │ 3.13.5 | 3.13.7
  │    │  │  │  └─ python_abi 3.13.* *_cp313, which cannot be installed
  │ because there are no viable options:
  │    │  │  │     └─ python_abi 3.13 | 3.13 | 3.13 | 3.13 | 3.13 would
  │ constrain
  │    │  │  │        └─ python 3.13.* *_cpython, which conflicts with any
  │ installable versions previously reported
  │    │  │  ├─ causal-conv1d 1.5.2 | 1.5.2 | 1.5.2 | 1.5.2 | 1.5.2 would
  │ require
  │    │  │  │  └─ python >=3.12,<3.13.0a0, which cannot be installed because
  │ there are no viable options:
  │    │  │  │     └─ python 3.12.0 | 3.12.1 | 3.12.1 | 3.12.2 | 3.12.3 |
  │ 3.12.4 | 3.12.5 | 3.12.6 | 3.12.6 | 3.12.6 | 3.12.7 | 3.12.8 | 3.12.9 |
  │ 3.12.9 | 3.12.10 | 3.12.11 would constrain
  │    │  │  │        └─ python_abi 3.12.* *_cp312, which conflicts with any
  │ installable versions previously reported
  │    │  │  ├─ causal-conv1d 1.5.2 | 1.5.2 | 1.5.2 | 1.5.2 | 1.5.2 would
  │ require
  │    │  │  │  └─ python >=3.11,<3.12.0a0, which cannot be installed because
  │ there are no viable options:
  │    │  │  │     └─ python 3.11.0 | 3.11.0 | 3.11.0 | 3.11.0 | 3.11.1 |
  │ 3.11.2 | 3.11.3 | 3.11.4 | 3.11.5 | 3.11.6 | 3.11.7 | 3.11.7 | 3.11.8 |
  │ 3.11.9 | 3.11.10 | 3.11.10 | 3.11.10 | 3.11.10 | 3.11.11 | 3.11.11 |
  │ 3.11.12 | 3.11.13 would constrain
  │    │  │  │        └─ python_abi 3.11.* *_cp311, which conflicts with any
  │ installable versions previously reported
  │    │  │  └─ causal-conv1d 1.5.2 | 1.5.2 | 1.5.2 would require
  │    │  │     ├─ libtorch >=2.7.1,<2.8.0a0, which cannot be installed
  │ because there are no viable options:
  │    │  │     └─ cuda-version >=12.9,<13, which cannot be installed because
  │ there are no viable options:
  │    │  │        └─ cuda-version 12.9 would constrain
  │    │  │           └─ cudatoolkit ==12.9|12.9.*, which conflicts with any
  │ installable versions previously reported
  │    │  └─ python_abi 3.10.* *_cp310, which cannot be installed because
  │ there are no viable options:
  │    │     ├─ python_abi 3.10 | 3.10 | 3.10 | 3.10 | 3.10 | 3.10 would
  │ constrain
  │    │     │  └─ python 3.10.* *_cpython, which conflicts with any
  │ installable versions previously reported
  │    │     └─ python_abi 3.10 would require
  │    │        └─ python 3.10.*, which cannot be installed because there are
  │ no viable options:
  │    │           ├─ python 3.10.18, which conflicts with the versions
  │ reported above.
  | ...
  │    │           └─ python 3.10.8, which conflicts with the versions
  │ reported above.
  │    └─ mamba-ssm 2.2.5 is excluded because due to strict channel priority
  │ not using this option from: 'file:///home/conda/feedstock_root/
  │ build_artifacts/'
```

</details>

forgot that the `triton <3.2.0` would cause pytorch 2.5 to be pulled in, and then we don't get `causal-conv1d==1.5.2` which is pinned to pytorch 2.7.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

weiji14 commented on recipe/recipe.yaml:
```diff
@@ -0,0 +1,69 @@
+extra:
+  recipe-maintainers:
+    - timkpaine
+    - weiji14
```
Added myself to the recipe maintainer list if you don't mind! I'd be interested in adding aarch builds later for Grace Hopper systems (xref https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/7764). Can also help with the CUDA 12.9 migration at #1 later.

weiji14 committed:
Add causal-conv1d optional runtime dependency

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint --conda-forge .` from the recipe directory. You can also examine the [workflow logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17636220549) for more detail.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/17636220549](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17636220549). Examine the logs at this URL for more detail.</sub>

weiji14 committed:
Unpin triton<3.2.0, disable python import check

Test would fail because of no GPU on CI server, but it's actually a bug that was fixed in https://github.com/triton-lang/triton/pull/6570 and would require triton>=3.4.0.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

weiji14 committed:
Free disk space on Azure pipelines

Xref https://conda-forge.org/docs/maintainer/conda_forge_yml/#azure

weiji14 committed:
MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.11.23.09.59

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

weiji14 commented:
@timkpaine / @isuruf, ok to give this another review?

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

I considered the following status checks when analyzing this PR:
 - **linter**: passed
 - **azure**: passed


Thus the PR was passing and merged! Have a great day!
